lib/lita/adapters/slack/api.rb

```diff
@@ -81,17 +93,21 @@ def set_topic(channel, topic)
           call_api("channels.setTopic", channel: channel, topic: topic)
         end
 
-        def rtm_start
-          response_data = call_api("rtm.start")
+        def rtm_connect
+          response_data = call_api("rtm.connect")
+          channels = conversations_list('public_channel,private_channel,im')
+
+          users = SlackUser.from_data_array(users_list)
+          user = users.find { |user| user.id == response_data['self']['id'] }
 
           TeamData.new(
-            SlackIM.from_data_array(response_data["ims"]),
-            SlackUser.from_data(response_data["self"]),
+            SlackIM.from_data_array(channels),
+            user,
             # Note: While membership info in each channel has been truncated at 500 users
             # since March 2018, this |users| array should still be complete.
-            SlackUser.from_data_array(response_data["users"]),
-            SlackChannel.from_data_array(response_data["channels"]) +
-              SlackChannel.from_data_array(response_data["groups"]),
+            users,
+            SlackChannel.from_data_array(channels) +
+              SlackGroup.from_data_array(channels),
```

Comment:
@jeremy-w Thinking through this...

We were previously joining an array of `channels` and `groups`.

Now that we're going to use `conversations.list` via
```rb
channels = conversations_list('public_channel,private_channel,im,mpim')
```

`channels` should have **all** channel data - so I'm thinking:
1. we don't even need the `SlackGroup` class
2. to join this array (since `SlackChannel.from_data_array(channels)` should have everything)
  * in fact, if we did join the array, I would expect duplicates in the resulting array.